### ISSUE TITLE
Commands mapped to parenthesis or brackets don't work on Nordic keyboards

### ISSUE THREAD

**Issue by NordicAngel** [helix-editor/helix#5718]
### Summary

On a Nordic keyboard (in my case danish but I think Sweden and Norway use the same layout)  there are problems with accessing commands that are mapped to parenthesis or brackets e.g. rotate_selections_backward ( or add_newline_below ]<space>. I think this is do to these characters being placed on the number keys i.e. I have to press shift-8 for an open parenthesis or alt gr-8 for a open bracket. If I try to use ( in normal mode it reads it as shift-( and I'm not even sure it can read my brackets. The commands switch_to_lowercase and switch_to_uppercase work even though I also need shift to access backtick, but they still read as shift-`.

I have gotten the reading by recording macros.

### Platform

Windows

### Terminal Emulator

windows cmd

### Helix Version

helix 22.12 (96ff64a8)

**Comment by CptPotato** [comment]
I'm using a German keyboard layout and faced similar issues with helix on windows. The parenthesis/brackets mapping seems to be the same as on your layout (<kbd>Shift</kbd><kbd>8</kbd> / <kbd>AltGr</kbd><kbd>8</kbd>). Using `(` / `)` to rotate selections works fine for me, but <kbd>AltGr</kbd> bindings register as <kbd>Ctrl</kbd><kbd>Alt</kbd><kbd>...</kbd>.

I managed to work around this, by adjusting some bindings. For example, here's the "unimpaired" bindings remapped:

```toml
"C-A-[" = { d = "goto_prev_diag", D = "goto_first_diag", f = "goto_prev_function", c = "goto_prev_class", a = "goto_prev_parameter", o = "goto_prev_comment", p = "goto_prev_paragraph", space = "add_newline_above", d = "goto_next_diag" }
"C-A-]" = { d = "goto_next_diag", D = "goto_last_diag", f = "goto_next_function", c = "goto_next_class", a = "goto_next_parameter", o = "goto_next_comment", t = "goto_next_test", t = "goto_prev_test", p = "goto_next_paragraph", space = "add_newline_below" }

```

**Comment by pascalkuthe** [comment]
As a german I highlight recommend switching away from nordic/german keyboards for CS they wrack your hands with they gymnastics you have to perform to type brackets.

That said it would be good to fix this in helix although I am not sure if that's possible. https://github.com/neovim/neovim/issues/11061 might be related

**Comment by CptPotato** [comment]
> As a german I highlight recommend switching away from nordic/german keyboards for CS they wrack your hands with they gymnastics you have to perform to type brackets.

I agree, especially `{[]}` with right <kbd>AltGr</kbd> is a mess, when using <kbd>Ctrl</kbd><kbd>Alt</kbd> it's not *that* bad. I actually bound the impaired keys to `ö/ä`. Getting used to all the special keys on a US layout would be a nightmare for me, so that's about the best I can do.
In the future I intend to switch to a [custom colemak layout](https://raw.githubusercontent.com/CptPotato/colemak_de/master/keymap.png) where I've duplicated some awkward to type characters to AltGr + left hand, though I haven't tried this extensively.

Anyway, to get back on topic. I think switching to `termwiz` is being considered, which might fix the issue with `[` being picked up as `C-A-[`.

**Comment by the-mikedavis** [comment]
I don't have a keyboard to test it but #4939 may fix the detection for these on terminals that support the enhanced keyboard protocol.

**Comment by CptPotato** [comment]
> I don't have a keyboard to test it but #4939 may fix the detection for these on terminals that support the enhanced keyboard protocol.

I gave the `enable-keycode-disambiguation` branch a try, but it did not help with this specific issue in both Wezterm and Cmd (on Windows). It's possible that this "keyboard enhancement" feature is not available with those terminals.
